src/app/view/modules/object/object.component.ts

```diff
@@ -97,4 +104,39 @@ export class ObjectComponent implements OnChanges, OnInit {
     }
```

Comment:
It could be worth to think about putting the whole subscription into a seperate function `getResource`.
Then you could call it from both ngOnChanges and ngOnInit without doubling the code here, like so:

```
 ngOnChanges() {
         this.getResource(this.iri);
}

 ngOnInit() {
         this._route.params.subscribe((params: Params) => {
             let resIri = ( params['rid'] !== undefined ? params['rid'] : this.iri );
             this.getResource(resIri);
         });
  }

getResource(iri: string) {
      this._resourceService.getResource(iri)
                 .subscribe(
                     (result: ApiServiceResult) => {
                         let promises = jsonld.promises;
                         // compact JSON-LD using an empty context: expands all Iris
                         let promise = promises.compact(result.body, {});
 
                         promise.then((compacted) => {
 
                             let resources: ReadResourcesSequence = ConvertJSONLD.createReadResourcesSequenceFromJsonLD(compacted);

                             this.resource = resources;
 
                            this.requestIncomingResources(resources);

                         }, function (err) {
 
                             console.log("JSONLD of full resource request could not be expanded:" + err);
                         });
 
                         this.isLoading = false;
                     },
                     (error: ApiServiceError) => {
                         this.errorMessage = <any>error;
                         this.isLoading = false;
                     }
    );
}
```